src/workerd/util/sqlite-test.c++

```diff
@@ -1076,5 +1076,96 @@ KJ_TEST("SQLite onRollback") {
   }
 }
 
+KJ_TEST("SQLite prepareMulti") {
+  auto dir = kj::newInMemoryDirectory(kj::nullClock());
+  SqliteDatabase::Vfs vfs(*dir);
+  SqliteDatabase db(vfs, kj::Path({"foo"}), kj::WriteMode::CREATE | kj::WriteMode::MODIFY);
+
+  auto stmt = db.prepareMulti(SqliteDatabase::TRUSTED, kj::str(R"(
+    CREATE TABLE IF NOT EXISTS things (
+      id INTEGER PRIMARY KEY AUTOINCREMENT,
+      value INTEGER
+    );
+    INSERT INTO things(value) VALUES (123);
+    INSERT INTO things(value) VALUES (456);
```

Comment:
Would there be value in testing a multi statement that fails partway through?  I guess it wouldn't make a difference at construction time, since it's lazy.  And I guess it would just throw the appropriate exception at run time?

Reply:
Good call. I think the `prelude` vector would end up incorrect... the queries before the one that failed would get duplicated on the next attempt.